2-projects/1-todo-list/script.js

```diff
@@ -12,14 +30,58 @@ let todos = [
 
 populateTodoList(todos);
 
+function createDOMListItem(todo) {
+  const li = document.createElement("li");
+
+  const tickIcon = document.createElement("i");
+  tickIcon.classList.add("fa", "fa-check");
+  const binIcon = document.createElement("i");
+  binIcon.classList.add("fa", "fa-trash");
+
+  if (!tickIcon.onclick) {
```

Comment:
Please note that `tickIcon.onclick` would still be `null` even if you have called `tickIcon.addEventListener()`. 

To make the condition valid, you can either assign onclick handler function directly:

```
tickIcon.onclick = () => {
      toggleIsComplete(todo);
      redrawTodo(li, todo.completed);
}
```

Or you can set an attribute on tickIcon to indicate if a listener is attached:
```
if (tickIcon.getAttribute('listener') !== 'true') {
    tickIcon.addEventListener('click', function (e) {
        const elementClicked = e.target;
        elementClicked.setAttribute('listener', 'true');
        toggleIsComplete(todo);
        redrawTodo(li, todo.completed);
    });
}
```

For details, you can have a look on https://stackoverflow.com/questions/11455515/how-to-check-whether-dynamically-attached-event-listener-exists-or-not